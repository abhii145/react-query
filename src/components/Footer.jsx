import { Link } from "react-router"

const Footer = () => {
  const currentYear = new Date().getFullYear()

  return (
    <footer className="bg-gray-800 text-white py-4 text-center">
      <div className="container mx-auto">
        <p>&copy; {currentYear} Your Company. All rights reserved.</p>
        <p>
          <Link to="/" className="text-blue-400 hover:text-blue-300">
            Privacy Policy
          </Link>
          {" | "}
          <Link to="/" className="text-blue-400 hover:text-blue-300">
            Terms of Service
          </Link>
        </p>
      </div>
    </footer>
  )
}

export default Footer
