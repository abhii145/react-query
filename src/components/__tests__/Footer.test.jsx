import { render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import Footer from "../Footer"
import { it, expect } from "vitest"

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Footer />
    </MemoryRouter>
  )
  expect(asFragment()).toMatchSnapshot()
})

it("renders Footer component", () => {
  render(
    <MemoryRouter>
      <Footer />
    </MemoryRouter>
  )

  expect(
    screen.getByText(/Your Company. All rights reserved./i)
  ).toBeInTheDocument()
  expect(
    screen.getByRole("link", { name: /Privacy Policy/i })
  ).toBeInTheDocument()
  expect(
    screen.getByRole("link", { name: /Terms of Service/i })
  ).toBeInTheDocument()
})
