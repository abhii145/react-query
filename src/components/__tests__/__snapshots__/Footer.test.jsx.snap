// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="bg-gray-800 text-white py-4 text-center"
  >
    <div
      class="container mx-auto"
    >
      <p>
        Â© 2024 Your Company. All rights reserved.
      </p>
      <p>
        <a
          class="text-blue-400 hover:text-blue-300"
          data-discover="true"
          href="/"
        >
          Privacy Policy
        </a>
         |
        <a
          class="text-blue-400 hover:text-blue-300"
          data-discover="true"
          href="/"
        >
          Terms of Service
        </a>
      </p>
    </div>
  </footer>
</DocumentFragment>
`;
